### PR TITLE
Fix archive storage not querying old spans older than maxSpanAge

### LINKED ISSUE #1511: Archived traces 404ing with elasticsearch storage
## Problem - what in Jaeger blocks you from solving the requirement?

With archive storage configured, I am getting 404s when trying to view an archived trace in the UI.

I am running Jaeger 1.11 with elasticsearch storage, version 6.2
The UI has a config w/ archiveEnabled set to true, `--es-archive.enabled=true`, and  `ES_SERVER_URLS` and `ES_ARCHIVE_SERVER_URLS` pointing to the same cluster.

I can successfully archive a trace in the UI and can see it stored in the jaeger-span-archive index.

However, accessing a trace via direct lookup at "trace/TRACE_ID" gives a 404.

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #1511

### CODE REVIEW

File: plugin/storage/es/spanstore/reader.go [hunk 1]
```diff
@@ -305,6 +321,7 @@ func (s *SpanReader) multiRead(ctx context.Context, traceIDs []model.TraceID, st
 	indices := s.timeRangeIndices(s.spanIndexPrefix, startTime.Add(-time.Hour), endTime.Add(time.Hour))
 	nextTime := model.TimeAsEpochMicroseconds(startTime.Add(-time.Hour))
 
+	fmt.Println(indices)
```
Comment (objectiser) [inline plugin/storage/es/spanstore/reader.go]: Still required?

File: plugin/storage/integration/elasticsearch_test.go [hunk 2]
```diff
@@ -99,14 +102,16 @@ func (s *ESStorageIntegration) initSpanstore(allTagsAsFields bool) {
...
 		})
 	s.SpanReader = spanstore.NewSpanReader(spanstore.SpanReaderParams{
 		Client:            client,
 		Logger:            s.logger,
 		MetricsFactory:    metrics.NullFactory,
 		IndexPrefix:       indexPrefix,
-		MaxSpanAge:        72 * time.Hour,
+		MaxSpanAge:        maxSpanAge,
 		TagDotReplacement: tagKeyDeDotChar,
+		Archive: archive,
```
Comment (objectiser) [inline plugin/storage/integration/elasticsearch_test.go]: nit: formatting looks wrong here

Comment (pavolloffay) [reply plugin/storage/integration/elasticsearch_test.go]: The check passed fine

File: plugin/storage/es/spanstore/reader.go [hunk 3]
```diff
@@ -159,6 +161,20 @@ func getTimeRangeIndexFn(archive, useReadWriteAliases bool) timeRangeIndexFn {
...
 
+func getSourceFn(archive bool, maxNumSpans int) sourceFn {
+	return func(query elastic.Query, nextTime uint64) *elastic.SearchSource {
+		s := elastic.NewSearchSource().
+			Query(query).
+			Size(defaultDocCount).
+			TerminateAfter(maxNumSpans).
+			Sort("startTime", true)
+		if !archive {
+			s.SearchAfter(nextTime)
```
Comment (pavolloffay) [inline plugin/storage/es/spanstore/reader.go]: This query uses scroll API https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-search-after.html.

`SearchAfter` says to include results with `startTime` parameter after `nextTime`. In archive storage we want to include all results.

**Pull request merged**